### DISCUSSION TITLE
Doesn't work

### DISCUSSION THREAD

**Discussion by nordseo** [KosmosisDire/obsidian-webpage-export#519] [General]
Out of the Box:

1. Doesn't generate URLs properly in the navbar.
2. Doesn't store media files in a media folder properly.
3. Doesn't have an Index page.
4. Doesn't have a Footer Colophon
5. Graph doesn't work.
6. Doesn't create child pages.

It has potential, but the most essential features to make a website don't work. It would have been wise to focus on functionality, not adding JS and CSS libs.

**Comment by KosmosisDire** [comment]
Have you tried the beta version? The beta version is currently way ahead of the stable version. And the dev branch is again further ahead. All of these are valid things to point out if you're having problems, but you haven't given any information that could help me reproduce the problems. If you'd like to test on the beta version and let me know which things are still a problem maybe I can actually do something about it

**Comment by KosmosisDire** [comment]
As far as specifics:
> Doesn't generate URLs properly in the navbar.

In what way are they improper?

> Doesn't store media files in a media folder properly.

Files are only stored in the media folder if they cannot be stored in their original location in the vault. It's a fallback location.

> Doesn't have an Index page.

You just create an index page by creating a file called index anywhere you want one. You can set the display title of these pages with the "title" frontmatter property. In the future I would like this to be easier by letting the user select which pages should be index pages

> Doesn't have a Footer Colophon

Not having a footer colophon is a feature request not a bug, however in the mean time you can easily add this using the custom html content. (I assume you are familiar with web dev based on this post)

> Graph doesn't work

Hard to tell why without more details, but trying the beta version is likely to help (there are currently some issues with firefox in the beta though, which will be fixed in the next beta release)

> Doesn't create child pages.

If you mean transclusions, this is fixed in the dev branch and will be released shortly

> It would have been wise to focus on functionality, not adding JS and CSS libs.

This sounds a bit passive aggressive but I'll take it haha. What do you mean by "adding JS and CSS libs" though?

**Comment by nordseo** [comment]
Thanks for the reply. I like the approach you are using, as most Obsidian website generators seem to be overly complex. 

I was willing to give another attempt today; however, when executing the export to HTML, the modal window is blank. 

I tried to uninstall, reinstall, and reboot, but it's still blank. (I'll give the beta a try.)

- I would prefer a media library instead of embedded data URI images.
- I haven't checked into the core, but I definitely prefer to build templates from scratch regarding CSS & JS libraries.

**Reply by KosmosisDire** [reply]
#520 The modal being blank I think is this issue. I am pretty confident this is fixed in the beta version.

> I definitely prefer to build templates from scratch regarding CSS & JS libraries.

What do you mean by building a template? I think I might be missing something haha

**Comment by nordseo** [comment]
Ah, sorry. One of the files had a misplaced backtick and broke a code block.

The links work suddenly, but I'm unsure how to replicate the issue.

Media is also loading from the `/media` folder, but I don't know what changed to make that work.

I will experiment, and if I have any issues, I'll investigate as much as possible before creating an issue.

**Reply by KosmosisDire** [reply]
(Embedded URIs only happens when you export a local site instead of a hosted site)
